test/StringUtilsTest.cpp

```diff
@@ -292,31 +299,34 @@ TEST(StringUtilsTest, insertThousandSeparator) {
 
   // Set the `floatingPointSignifier` to characters, that are reserved regex
   // characters.
-  auto reservedRegexCharTest = [&doNotExceptionTest]<const char... Cs>() {
-    (doNotExceptionTest.template operator()<Cs>(' '), ...);
-  };
+  auto reservedRegexCharTest = ad_utility::ApplyAsValueIdentity{
+      [&doNotExceptionTest](auto... valueIdentities) {
+        (doNotExceptionTest.template operator()<valueIdentities.value>(' '),
+         ...);
+      }};
   reservedRegexCharTest
       .template operator()<'.', '(', ')', '[', ']', '|', '{', '}', '*', '+',
                            '?', '^', '$', '\\', '-', '/'>();
 
   // Numbers as `separatorSymbol`, or `floatingPointSignifier`, are not allowed.
-  auto forbiddenSymbolTest =
-      [&doNotExceptionTest]<const char... floatingPointSignifiers>() {
+  auto forbiddenSymbolTest = ad_utility::ApplyAsValueIdentity{
+      [&doNotExceptionTest](auto... floatingPointSignifiers) {
         for (size_t separatorSymbolNum = 0; separatorSymbolNum < 10;
              separatorSymbolNum++) {
           const char separatorSymbol{absl::StrCat(separatorSymbolNum).front()};
-          (
-              [&doNotExceptionTest, &separatorSymbol]<const char c>() {
-                ASSERT_ANY_THROW(
-                    doNotExceptionTest.template operator()<c>(' '));
-                ASSERT_ANY_THROW(doNotExceptionTest.template operator()<'.'>(
-                    separatorSymbol));
-                ASSERT_ANY_THROW(
-                    doNotExceptionTest.template operator()<c>(separatorSymbol));
-              }.template operator()<floatingPointSignifiers>(),
-              ...);
+          (ad_utility::ApplyAsValueIdentity{[&doNotExceptionTest,
+                                             &separatorSymbol](
+                                                auto valueIdentity) {
+             static constexpr char c = valueIdentity.value;
+             ASSERT_ANY_THROW(doNotExceptionTest.template operator()<c>(' '));
+             ASSERT_ANY_THROW(
+                 doNotExceptionTest.template operator()<'.'>(separatorSymbol));
+             ASSERT_ANY_THROW(
+                 doNotExceptionTest.template operator()<c>(separatorSymbol));
+           }}.template operator()<floatingPointSignifiers.value>(),
+           ...);
         }
-      };
+      }};
```

Comment:
Here we can get rid of the `.template operator()` as well as the `ApplyAsValueIdentity`.